spec/execution_tests.lua

```diff
@@ -1749,6 +1749,71 @@ function execution_tests.run(compile_file, backend, _ENV, only_compile)
         end)
     end)
 
+    describe("For-in loops", function()
+        compile([[
+            local function iter(state: any, current: any): (any, any)
+                if current as integer == #(state as {any}) then
+                    return nil, nil
+                end
+
+                return current as integer + 1, (state as {any})[current as integer + 1]
+            end
+
+            typealias iterfn = (any, any) -> (any, any)
+            local function xpairs(xs: {any}): (iterfn, any, any)
+                return iter, xs, 0
+            end
+
+
+            export function double(xs: {integer}): {integer}
+                local out: {integer} = {}
+                for i, x in xpairs(xs) do
+                    out[#out + 1] = x as integer * 2
+                end
+
+                return out
+            end
+
+            export function abs_list(xs: {float})
+                for i, x in xpairs(xs) do
+                    if (x as float) < 0.0 then
+                        xs[i] = -(x as float)
+                    end
+                end
+            end
+        ]])
+
+        it("", function()
+            run_test([[
+                local xs = test.double({1, 2})
+                assert(xs[1] == 2 and xs[2] == 4)
+
+                xs = {0.0, 2.1, -1.0, -4.5}
+                test.abs_list(xs)
+                assert(xs[1] == 0.0 and xs[2] == 2.1 and xs[3] == 1.0 and xs[4] == 4.5)
+            ]])
+        end)
+    end)
+
+    describe("Nested for-in loops", function()
+        compile([[
+            export function iter(a: any, b: any): (any, any)
+                return 1, 2
+            end
+
+            export function fn()
+                local a: any = 2
+                local b: any = 3
+                for i, j in iter, a, b do
+                    for k, v in iter, b, a do
+                        i = v
+                        k = j
+                    end
+                end
+            end
+        ]])
```

Comment:
This test case is missing an `it` block.  My suggestion would be to rewrite this test case to use that xpairs/my_ipairs function from before, because it's easier to think of an example if we use that. We can put it next to the other `my_ipairs` test cases.